src/main/kotlin/com/intershop/gradle/icm/docker/utils/AbstractTaskPreparer.kt

```diff
@@ -50,63 +53,157 @@ abstract class AbstractTaskPreparer(
 
-    protected fun configureContainerTask(task: DockerCreateContainer) {
-        task.group = "icm container ${getContainerExt()}"
-        task.attachStderr.set(true)
-        task.attachStdout.set(true)
-        task.hostConfig.autoRemove.set(true)
-
-        task.containerName.set(getContainerName())
-
-        val os: OperatingSystem = DefaultNativePlatform.getCurrentOperatingSystem()
-        if(useHostUser() && !os.isWindows) {
-            val system = com.sun.security.auth.module.UnixSystem()
-            val uid = system.uid
-            val userName = system.username
-            val gid = system.gid
-            project.logger.info("Using user {}({}:{}) to start container {}", userName, uid, gid, getContainerName())
-            task.user.set(uid.toString())
+    /**
+     * Same as [registerCreateContainerTask] using `taskType == CreateExtraContainer`. `volumes` and `env` are wrapped into [Provider]s.
+     */
+    protected fun registerCreateContainerTask(
+            findTask: TaskProvider<FindContainer>, volumes: Map<String, String>,
+            env: ContainerEnvironment,
+    ): TaskProvider<CreateExtraContainer> {
+        return registerCreateContainerTask(findTask, CreateExtraContainer::class.java, project.provider { volumes },
+                project.provider { env })
+    }
+
+    /**
+     * Registers the task that creates the container
+     * @param findTask a [TaskProvider] pointing to the [FindContainer]-task
+     * @param taskType the actual task type to be created
+     * @param volumes a [Provider] for the volumes to be bound. Local directories are created on demand.
+     * @param env a [Provider] for the container environment to be used
+     * @return a [TaskProvider] pointing to the registered task
+     */
+    protected fun <T> registerCreateContainerTask(
+            findTask: TaskProvider<FindContainer>, taskType: Class<T>,
+            volumes: Provider<Map<String, String>>,
+            env: Provider<ContainerEnvironment>,
+    ): TaskProvider<T> where T : CreateExtraContainer {
+        return project.tasks.register(taskNameOf("create"), taskType) { task ->
+            task.group = taskGroup
+            task.description = "Creates the ${getContainerExt()}-container is not already existing"
+            task.attachStderr.set(true)
+            task.attachStdout.set(true)
+            task.hostConfig.autoRemove.set(true)
+
+            task.targetImageId(project.provider { pullTask.get().image.get() })
+            task.image.set(project.provider { pullTask.get().image.get() })
+
+            task.hostConfig.network.set(networkId)
+            task.withEnvironment(env)
+
+            task.dependsOn(findTask, networkTask)
+            task.existingContainer.set(project.provider { findTask.get().foundContainer.orNull })
+            task.containerName.set(getContainerName())
+
+            val os: OperatingSystem = DefaultNativePlatform.getCurrentOperatingSystem()
```

Comment:
I think it would be better to rely on the system property `System.properties['os.name']` for this. Because `DefaultNativePlatform` is internal, it may break with any minor gradle release.